pkg/liqo-controller-manager/resource-request-controller/resourceRequest_controller.go

```diff
@@ -122,17 +123,28 @@ func (r *ResourceRequestReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 	}()
 
+	var requeue bool
 	// ensure creation, update and deletion of the related ResourceOffer
 	switch resourceReqPhase {
 	case allowResourceRequestPhase:
 		// ensure that we are offering resources to this remote cluster
-		r.Broadcaster.enqueueForCreationOrUpdate(remoteClusterID)
+		requeue, err = r.OfferUpdater.CreateOrUpdateOffer(remoteCluster)
+		if err != nil {
+			klog.Errorf("Error creating a ResourceOffer: %s", err)
+			if requeue {
+				return ctrl.Result{
+					Requeue:      true,
+					RequeueAfter: 1 * time.Second,
+				}, err
+			}
```

Comment:
A requeue is automatically performed in case of errors. Hence, this is not required.